src/sorting/BubbleSort.sol

```diff
@@ -22,6 +22,28 @@ library BubbleSort {
         }
     }
 
+    /**
+     * @dev sort an uint256 array in place with bubble sort and return indexes
+     */
+    function argSort(uint256[] memory arr) internal pure returns (uint256[] memory idxs) {
+        idxs = new uint256[](arr.length);
+        unchecked {
+            uint256 i;
+            for (i; i < arr.length; ++i) {
+                idxs[i] = i;
+            }
+            if (arr.length <= 1) return (idxs);
+            for (i = 0; i < arr.length; ++i) {
```

Comment:
felt like you can merge these 2 loops:
just specify `idxs[i] = i` before the inner j loop

Reply:
I made argSort indirect (not in-place sort) in order to preserve the original array. So it know need this outer loop.